src/vs/vscode.proposed.d.ts

```diff
@@ -168,14 +168,19 @@ declare module 'vscode' {
 		availableFormats: ColorFormat[];
 
 		/**
+		 * Controls whether the color decorator is rendered.
+		 */
+		renderDecorator: boolean;
```

Comment:
I'm convinced, will move the config to the core. I only have one question

> we control the UI and the extensions just provide the data

Is it possible/reasonable that the provider provides view state? AFAIK, we already did that in TreeDataProvider, which feeds us `TreeItem` and `TreeItem` has a view state `collapsibleState` (whether it's collapsed or expanded). In our case, `renderDecorator` decides the visibility of the inline color decorator. @jrieken thoughts?